elements/Shape.js

```diff
@@ -10,6 +11,17 @@ class Shape extends Component {
         });
         this.state = this.touchableGetInitialState();
     }
+
+    setNativeProps = (args) => {
+        var extracted = extractProps(args, this)
```

Comment:
I see.  

I'm not sure the goal of setNativeProps is to act as a bypass like this.  I believe it's meant to be used to skip the react rendering flow for efficiency reasons alone.  If that's the case (please correct me if I'm wrong), then you would typically hope to see the same rendering behavior with regular props as well as setNativeProps.  

The problem that I was trying to solve for my use case is that initial rendering worked great, but then i could not change the value via setNativeProps (in my case for animating the fill color).  This PR fixes that for me and makes things consistent between the initial render and setNativeProps changes.  

It seems that you were working around some logic in this extractProps() function, when maybe the correct fix would be for us to fix that logic or add an explicit bypass for matrices there in addition to my changes?  Can we somehow change extractProps() to accept a matrix in the format you want?  

I'm not sure how many other people are working with your same expectation however.  It might break many people's code?